src/transactionWatcher.ts

```diff
@@ -155,6 +161,10 @@ export class TransactionWatcher {
             } catch (error) {
                 Logger.debug("TransactionWatcher.awaitConditionally(): cannot (yet) fetch data.");
 
+                if (error instanceof ErrIsCompletedFieldIsMissingOnTransaction) {
```

Comment:
This is correct.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#conditional_catch_blocks

@CiprianDraghici can confirm, as well.